bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+Likewise, the _input script_ is a transaction field, but the length indicator of the input script a serialization artifact.
+
+===Definition of Terms===
+
+; Annex
+: [TC] A Witness Item reserved for future extensions, currently unused (see [[https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-5|BIP341]]).
+
+; Amount
+: [TC] The count of satoshis assigned to an output.
+
+; Bare Output Script
+: [AT] A Condition Script that is directly defined in the Output Script field. Used to specify all types of Bare Outputs including Pay to Pubkey, Pay to Public Key Hash, Pay to Multisig outputs, and arbitrary scripts.
+
+; Condition Script
+: [AT] Collective term for scripts that are not forwarding scripts and encode a set of conditions that could be fulfilled to spend a UTXO.  A scriptPubkey can either contain directly a condition script in the case of a bare output, or will commit to one or multiple condition scripts for other output types.  For P2TR the tweaked public key is considered a condition script as well as each Leaf Script in the script tree.  At the time of writing, there are four distinct Condition Scripts: Witness Script, Leaf Script, Redeem Script (only non-segwit P2SH), and Bare Output Script (bare outputs, Pay to Multisig, P2PK, P2PKH).
+
+; Control Block
+: [TC] The proof of existence for a Leaf Script.  Appears as a Witness Item in Witness Stacks for P2TR script path inputs.  Only consists of the Inner Key for Script Trees with a single Leaf Script, otherwise provides the Inner Key as well as a Taproot Branch in the form of the hashing partners for the used Leaf Script.
+
+; Flag
+: [SA] Serialization artifact indicating features used by the transaction.  As of writing, the only allowed value is <code>01</code> which indicates that the transaction serialization has a Witness Section (see BIP144).
+
+; Forwarding Script
+: [AT] A collective term for scripts that redirect input validation to another script or data structure.  Witness Programs and P2SH Programs are forwarding scripts.
+
+; Input Counter
+: [SA] The length of the transaction input list.
+
+; Input Index
+: [AT] The position of an input in a transaction's Input List.
+
+; Input Script
+: [TC] originally called `scriptSig`. The Bitcoin code base used Hungarian notation when it was published.  This transaction field appearing in Transaction Inputs was named for containing the signature and being of the type script.  Today, this field may be better understood as the _input script_ which is used to satisfy spending conditions for non-segwit outputs, redirects validation to the Witness Stack for wrapped segwit outputs, and is empty for native segwit outputs. The Input Script may contain an actual script when spending a Bare Output Script, but may only contain push operations for any output type since P2SH.
```

Comment:
"the _input script_ which is used to satisfy spending conditions for non-segwit outputs, redirects validation to the Witness Stack for wrapped segwit outputs, and is empty for native segwit outputs."

This sentence clarifies A LOT of confusion I have had in the past about this TC!